pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -198,7 +198,28 @@ func extractLibInfo(pod *corev1.Pod, containerRegistry string) []libInfo {
 				})
 			}
 		}
+	}
 
+	if len(libInfoList) == 0 {
+		// Inject all if admission.datadoghq.com/all-lib.version exists
+		// without any other language-specific annotations.
+		// This annotation is typically expected to be set via remote-config
+		// for batch instrumentation without language detection.
+		injectAllAnnotation := strings.ToLower(fmt.Sprintf(libVersionAnnotationKeyFormat, "all"))
+		if version, found := podAnnotations[injectAllAnnotation]; found {
+			// This logic will be updated once we bundle all libs in
+			// one single init container. Versions will be supported by then.
+			if version != "latest" {
+				log.Warnf("Ignoring version %q. To inject all libs, the only supported version is latest for now", version)
+				version = "latest"
+			}
```

Comment:
should we add a "warning" annotation on the pod. to the limitation is more visible for the user? 
I'm worry that the user will not look at the cluster-agent logs to understand why the version set is not taken into account.

Reply:
Or maybe we can think of adding the capability to the admission controller to create `k8s events`  to attach to the new pod the info.

Reply:
Just to be clear, we won't ask users to set this annotation and won't document it. It will be used by RC, so version validation also happens on the UI and the backend.
Generally speaking, I really like the idea of attaching k8s events!

Reply:
I'm also happy to throw an error here and not default to `latest`, I think it'd be safer and it means the BE sent invalid configs. Wdyt?

Reply:
Closing this thread as we have ideas to better address this in the future as discussed offline.